reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -2100,6 +2101,33 @@ public static <T, D> Flux<T> using(Callable<? extends D> resourceSupplier, Funct
 		return using(resourceSupplier, sourceSupplier, resourceCleanup, true);
 	}
 
+	/**
+	 * Uses a {@link AutoCloseable} resource, generated by a supplier for each individual Subscriber,
+	 * while streaming the values from a Publisher derived from the same resource and makes sure
+	 * the resource is released if the sequence terminates or the Subscriber cancels.
+	 * <p>
+	 * Eager {@link AutoCloseable} resource cleanup happens just before the source termination and exceptions raised
+	 * by the cleanup Consumer may override the terminal event.
+	 * <p>
+	 * <img class="marble" src="doc-files/marbles/usingForFlux.svg" alt="">
+	 * <p>
+	 * For an asynchronous version of the cleanup, with distinct path for onComplete, onError
+	 * and cancel terminations, see {@link #usingWhen(Publisher, Function, Function, BiFunction, Function)}.
+	 *
+	 * @param resourceSupplier a {@link Callable} that is called on subscribe to generate the resource
+	 * @param sourceSupplier a factory to derive a {@link Publisher} from the supplied resource
+	 * @param <T> emitted type
+	 * @param <D> resource type
+	 *
+	 * @return a new {@link Flux} built around a disposable resource
+	 * @see #usingWhen(Publisher, Function, Function, BiFunction, Function)
+	 * @see #usingWhen(Publisher, Function, Function)
+	 */
+	public static <T, D extends AutoCloseable> Flux<T> using(Callable<D> resourceSupplier,
```

Comment:
I do feel we also need the counter-part that allows to delay the resource cleanup until after the sequence has terminated (using the `eager` argument).

Reply:
✅ updated, add method with `eager` argument!